core/src/main/scala/flow/Generator.scala

```diff
@@ -2,137 +2,329 @@ package datto.flow
+      Generator.Mat.future(Future.successful(source), maybeOptions)
   }
 
-  def future[T](sourceBuilder: ⇒ Future[Source[T, akka.NotUsed]])(implicit ec: ExecutionContext): Generator[T, Unit] =
-    new Generator(() ⇒ sourceBuilder.map(toUnit))
-
-  def apply[T](source: Source[T, akka.NotUsed])(implicit ec: ExecutionContext): Generator[T, Unit] =
-    Generator.Mat(toUnit(source))
-
-  def apply[T](stream: Stream[T])(implicit ec: ExecutionContext): Generator[T, Unit] =
-    Generator.Mat(toUnit(Source(stream)))
-
-  def empty[T](implicit ec: ExecutionContext) = Generator[T](Source.empty[T])
-
-  def single[T](item: T)(implicit ec: ExecutionContext) = Generator[T](Source.single(item))
-
-  def iterator[T](it: () ⇒ Iterator[T])(implicit ec: ExecutionContext) =
-    Generator(Source.fromIterator(it))
-
-  def futureGenerator[T, Out](futureGen: ⇒ Future[Generator[T, Out]])(implicit ec: ExecutionContext) =
-    Generator.Mat.future(futureGen.flatMap(_.source()))
-
-  def failed[T, Out](e: Throwable) = Generator.Mat.future[T, Out](Future.failed[Source[T, Future[Out]]](e))
-
-  private def toUnit[T](source: Source[T, akka.NotUsed])(implicit ec: ExecutionContext): Source[T, Future[Unit]] =
+  def future[T](
+    sourceBuilder: ⇒ Future[Source[T, akka.NotUsed]],
+    maybeOptions: Option[GeneratorOptions] = None)(
+    implicit
+    ec: ExecutionContext): Generator[T, Unit] =
+    new Generator(() ⇒ sourceBuilder.map(toUnit), maybeOptions)
+
+  def apply[T](
+    source: Source[T, akka.NotUsed],
+    maybeOptions: Option[GeneratorOptions])(
+    implicit
+    ec: ExecutionContext): Generator[T, Unit] =
+    Generator.Mat(toUnit(source), maybeOptions)
+
+  def apply[T](source: Source[T, akka.NotUsed])(
+    implicit
+    ec: ExecutionContext): Generator[T, Unit] = apply(source, None)
+
+  def apply[T](stream: Stream[T], maybeOptions: Option[GeneratorOptions])(
+    implicit
+    ec: ExecutionContext): Generator[T, Unit] =
+    Generator.Mat(toUnit(Source(stream)), maybeOptions)
+
+  def apply[T](stream: Stream[T])(
+    implicit
+    ec: ExecutionContext): Generator[T, Unit] =
+    apply(stream, None)
+
+  def empty[T](implicit ec: ExecutionContext) =
+    Generator[T](Source.empty[T], None)
+
+  def single[T](item: T, maybeOptions: Option[GeneratorOptions] = None)(
+    implicit
+    ec: ExecutionContext) =
+    Generator[T](Source.single(item), maybeOptions)
+
+  def iterator[T](
+    it: () ⇒ Iterator[T],
+    maybeOptions: Option[GeneratorOptions] = None)(
+    implicit
+    ec: ExecutionContext) =
+    Generator(Source.fromIterator(it), maybeOptions)
+
+  def futureGenerator[T, Out](futureGen: ⇒ Future[Generator[T, Out]])(
+    implicit
+    ec: ExecutionContext): Generator[T, Out] =
+    //TODO: For code review, let's take an extra close look at this.  I _think_ I am not making it evaluate futureGen
+    //any sooner than it otherwise would have, but I want another set of eyes
```

Comment:
I agree with this. I'm not sure how intentional that is in the original code though, kind of seems like a bug. It's a good example why `() => T` should probably be preferred to `=> T`.  But, LGTM